openesdh-repo/src/main/java/dk/openesdh/repo/services/documents/DocumentCategoryServiceImpl.java

```diff
@@ -0,0 +1,133 @@
+package dk.openesdh.repo.services.documents;
+
+import dk.openesdh.repo.model.DocumentCategory;
+import dk.openesdh.repo.model.OpenESDHModel;
+import dk.openesdh.repo.services.system.OpenESDHFoldersService;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.alfresco.error.AlfrescoRuntimeException;
+import org.alfresco.model.ContentModel;
+import org.alfresco.service.cmr.repository.ChildAssociationRef;
+import org.alfresco.service.cmr.repository.InvalidNodeRefException;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.NodeService;
+import org.alfresco.service.namespace.QName;
+
+public class DocumentCategoryServiceImpl implements DocumentCategoryService {
+
+    public static List<String> SYSTEM_TYPES = Arrays.asList("annex");
+    private OpenESDHFoldersService openESDHFoldersService;
+    private NodeService nodeService;
+
+    @Override
+    public List<DocumentCategory> getDocumentCategories() {
+        return nodeService.getChildAssocs(openESDHFoldersService.getDocumentCategoriesRootNodeRef())
+                .stream()
+                .map(assocItem -> getDocumentCategory(assocItem.getChildRef()))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public DocumentCategory saveDocumentCategory(DocumentCategory documentCategory) {
```

Comment:
The method is a bit confusing. Perhaps it could be logically split into "createNew" and "update" parts or even separate methods to enhance readability.